### PR TITLE
Migrate SettingsSAMLForm to formik

### PR DESCRIPTION
**Pull request by shaunlebron**
Checking off an item from https://github.com/metabase/metabase/issues/26178

## Summary

1. Created `FormTextarea` in `metabase/forms`, which is just a copy of `FormTextInput` with a textarea
2. Created `FormSwitch` in `metabase/forms` for using a mantine Switch inside formik.
3. Modified `FormTextInput` to take an optional `hasCopyButton` prop (to replace `CopyWidget` when using formik)
4. Modified `GroupMappingsWidget` to use `FormSwitch`, but only if used in formik— to avoid breaking SettingsLdapForm and SettingsJWTForm.
5. Added a whitelist of setting keys and specify which ones are overridable by their defaults.
6. Migrate all form fields in `SettingsSAMLForm` to our `metabase/forms` components.

## How to verify

Run EE tests for `saml.cy.spec.js` and unit tests.

To test with actual SAML, use:

* https://github.com/paoliniluis/metabase-keycloak
* https://www.metabase.com/docs/latest/people-and-groups/saml-keycloak

### CODE REVIEW

File: enterprise/frontend/src/metabase-enterprise/auth/components/SettingsSAMLForm/SettingsSAMLForm.jsx [hunk 1]
```diff
@@ -82,12 +89,13 @@ const SettingsSAMLForm = ({ elements = [], settingValues = {}, onSubmit }) => {
...
-            <div className="Form-label">{t`URL the IdP should redirect back to`}</div>
-            <div className="pb1">{t`This is called the Single Sign On URL in Okta, the Application Callback URL in Auth0,
-                                  and the ACS (Consumer) URL in OneLogin. `}</div>
-            <CopyWidget value={getAcsCustomerUrl()} />
-          </div>
+          <FormTextInput
+            name={FAKE_ACS_URL_KEY}
+            description={t`This is called the Single Sign On URL in Okta, the Application Callback URL in Auth0, and the ACS (Consumer) URL in OneLogin. `}
+            hasCopyButton={true}
+            readOnly={true}
```
Comment (iethree) [inline enterprise/frontend/src/metabase-enterprise/auth/components/SettingsSAMLForm/SettingsSAMLForm.jsx:96]: we usually just use the react shorthand:
```
hasCopyButton
readOnly
```
which implies true

Comment (shaunlebron) [reply enterprise/frontend/src/metabase-enterprise/auth/components/SettingsSAMLForm/SettingsSAMLForm.jsx]: https://github.com/metabase/metabase/pull/35731/commits/0ce87ad2b29c3e8b4f50f690f1e7f4cafcf3536f

File: frontend/src/metabase/forms/components/FormTextarea/FormTextarea.tsx
Comment (iethree) [inline frontend/src/metabase/forms/components/FormTextarea/FormTextarea.tsx:1]: can you add unit tests for this?

File: frontend/src/metabase/forms/components/FormSwitch/FormSwitch.tsx
Comment (iethree) [inline frontend/src/metabase/forms/components/FormSwitch/FormSwitch.tsx:1]: can you add unit tests for this? you can probably copy most of them from `FormTextInput` and modify slightly.

File: frontend/src/metabase/forms/components/FormTextInput/FormTextInput.tsx [hunk 2]
```diff
@@ -3,15 +3,24 @@ import type { ChangeEvent, FocusEvent, Ref } from "react";
...
   nullable?: boolean;
+  hasCopyButton?: boolean;
 }
 
 export const FormTextInput = forwardRef(function FormTextInput(
-  { name, nullable, onChange, onBlur, ...props }: FormTextInputProps,
+  {
+    name,
+    nullable,
+    hasCopyButton,
```
Comment (iethree) [inline frontend/src/metabase/forms/components/FormTextInput/FormTextInput.tsx:19]: can you add unit test coverage for this new prop?

**Review by iethree — CHANGES_REQUESTED** [review]
Looks great! Can you just add unit test coverage for the new/modified form components? We want to make especially sure they are well covered since they're so foundational.

File: frontend/src/metabase/forms/components/FormTextInput/FormTextInput.tsx (hunk 2 above)
Comment (shaunlebron) [reply frontend/src/metabase/forms/components/FormTextInput/FormTextInput.tsx:19]: fd5f11a53413a5656d034192df9356bf53d4e27c

File: frontend/src/metabase/forms/components/FormTextarea/FormTextarea.tsx
Comment (shaunlebron) [reply frontend/src/metabase/forms/components/FormTextarea/FormTextarea.tsx:1]: 9250889daaafda48112008e70101feafb5cd326a

**Review by iethree — APPROVED** [review]
✨

**Comment by iethree** [comment]
I wonder if @likeshumidity or @paoliniluis has a setup that could validate this working with actual SAML?

**Review by npfitz — APPROVED** [review]
Looks good!

**Review by dragonsahead — APPROVED** [review]
LGTM

**Pull request merged**